src/main/java/me/hypherionmc/sdlinklib/discord/slashcommands/LinkSlashCommand.java

```diff
@@ -101,23 +100,21 @@ protected void execute(SlashCommandEvent event) {
             List<UserTable> tables = userTable.fetchAll("discordID = '" + event.getUser().getIdLong() + "'");
 
             if (tables.isEmpty()) {
-                event.reply("Your discord account does not appear to be linked to a minecraft account");
+                event.reply("Your discord account does not appear to be linked to a minecraft account").setEphemeral(true).queue();
             } else {
                 tables.forEach(SQLiteTable::delete);
 
-                String nickName = (event.getMember().getNickname() == null || event.getMember().getNickname().isEmpty()) ? event.getUser().getName() : event.getMember().getNickname();
-                if (pattern.matcher(nickName).matches()) {
-                    nickName = pattern.matcher(nickName).replaceAll("");
-                }
-
-                try {
-                    event.getMember().modifyNickname(nickName).queue();
-                } catch (Exception e) {
-                    if (modConfig.generalConfig.debugging) {
-                        e.printStackTrace();
+                String nickName = event.getMember().getEffectiveName();
+                if (pattern.matcher(nickName).find()) {
+                    try {
+                        event.getMember().modifyNickname(null).queue();
```

Comment:
Same as before, don't clear the nickname

Reply:
Same as before